### ISSUE TITLE
Previous global option prevents devtools::check() to run checks.

### ISSUE THREAD

**Issue by antaldaniel** [r-lib/testthat#835]
Following on [#775](https://github.com/r-lib/testthat/issues/775) I have a very subtle problem with running devtools::check()

A global option prevents devtools::check () to run, and even it gives a false positive report on the checks.

```
-- R CMD check ---------------------------------------------------------------------
   Error: 2:39: unexpected symbol
   2: options(blogdown.hugo.dir = 'C:/Hugo').Last
                                             ^
   Execution halted
-- R CMD check results ------------------------------------ eurobarometer 0.1.0 ----
Duration: 365ms

0 errors √ | 0 warnings √ | 0 notes √

```
I had a global option setting for my Hugo library, because, like many Windows users, I had problems with Windows 10 using space in paths, and put it to a high level C:/Hugo library.

I tried options ( blogdown.hugo.dir = NULL) but it did not help, nor starting a new R session. 
Then I tried settings::reset_options() still no avail.

Currently, my options do not include blogdown.hugo.dir but somehow this previous setting prevents devtools to run the checks. (I tried on a fresh package with three very simple functions.)

**Comment by jimhester** [comment]
`options(blogdown.hugo.dir = 'C:/Hugo').Last` is not valid R code, where are you setting this option?

**Comment by antaldaniel** [comment]
I was following the blogdown instructions, and I certainly had no intention to add .Last to it. 
[D1 More global options](https://bookdown.org/yihui/blogdown/more-global-options.html)

Checking with options() this is the global option

```
$blogdown.hugo.dir
[1] "C:/Hugo"
```

So I have no idea where the .Last is coming in from, but so far I had only problem with this devtools.

**Comment by yihui** [comment]
I don't know where `.Last` came from, but you need to make sure your .Rprofile has a newline at the end: https://yihui.name/en/2018/04/rprofile-trailing-newline/ (this has been emphasized in the **blogdown** book: https://bookdown.org/yihui/blogdown/global-options.html).

**Comment by jimhester** [comment]
Please try running `R CMD check` from the command line to isolate this outside of devtools.

**Comment by antaldaniel** [comment]
I think that @yihui was right, though I am not sure about the sequence of things, because both hugo.exe and devtools pointed to a different direction.  Hugo.exe first suggested a problem with whitespace in the in the Window10 path, but in fact, the whitespace problem was an unclosed " in an .md file. Moreover, a copy of the .Renviron line may have missed end of line character.  

Obviously it is my mistake all, and took a very long time to figure out the issue, but how it crept into devtools was a .Last parameter that is probably being added when before the checks as the .Rprofile and .Renviron variables are read in.  

My apologies if I made false alarm, the reason why I think that this may somehow be created the way devtools::check() works is that when I created a new session in r, obviously the .Renviron was read in without creating any problem (even though it may have been ignored, but it was anyway an irrelevant entry to this package.)

I wonder if there is a way to devtools signal if the error is specifically a setup error, pointing directly to .Rprofile and .Renviron?

(Just because of the timing Happy New Year and thanks for the help)
